packages/@react-stately/utils/src/useControlledState.ts

```diff
@@ -65,10 +67,12 @@ export function useControlledState<T>(
       setStateValue(updateFunction);
     } else {
       if (!isControlled) {
+        console.log('setValue: setting state clone and value', Object.assign({}, value), value);
         setStateValue(value);
       }
       onChangeCaller(value, ...args);
     }
+    console.log('did I end up here?', isControlled, Object.assign({}, value));
```

Comment:
These show the setDate/value update I expected, but it is called by something I'm not able to track to update with a different value at the same time. Late night thought that we could have two instances of the state object?